src/llvm/ir.rs

```diff
@@ -123,14 +100,23 @@ impl Metadata {
             | LLVMMetadataKind::LLVMDIStringTypeMetadataKind
             | LLVMMetadataKind::LLVMDIGenericSubrangeMetadataKind
             | LLVMMetadataKind::LLVMDIArgListMetadataKind
-            | LLVMMetadataKind::LLVMDIAssignIDMetadataKind => MetadataKind::Unknown(self),
+            | LLVMMetadataKind::LLVMDIAssignIDMetadataKind => Metadata::Unknown(value),
         }
     }
 }
 
+impl TryFrom<MDNode> for Metadata {
+    type Error = ();
+
+    fn try_from(md_node: MDNode) -> Result<Self, Self::Error> {
```

Comment:
Is this being used anywhere? I think your `metadata()` method from another place does it in more sane way (and is being used).

Reply:
metadata() returns LLVMMetadataRef, this returns Metadata and it's used here https://github.com/aya-rs/bpf-linker/blob/baafefaa27dd44171c9c9150c9ab18267812083f/src/llvm/di.rs#L543